particle-iot/device-os
# [p2] [wifi] allow setting common country code and channel plan in P2

<details>
  <summary><i>Enable setting of common wifi country codes from ISO 3166-1 alpha-2</i></summary>

</details>

### Problem

With the P2 previously we only supported setting the wifi country code to "US". 

### Solution

This PR allows setting some common country codes, along with our best understanding of the MCU-supported channel plan configuration to match these regions. 

### Steps to Test

Set country code to one of the supported versions. Verify that you're able to connect to a corresponding wifi access point in that region. 

### Example App

```c

SYSTEM_THREAD(ENABLED);
SYSTEM_MODE(SEMI_AUTOMATIC);
SerialLogHandler logHandler(115200, LOG_LEVEL_INFO);

void setup() {
    wlan_set_country_code("JP");
}

void loop() {
    if (!Particle.connected()) {
        Log.info("Particle connect...");
        delay(3000);
        Particle.connect();
        waitFor(Particle.connected, 30000);
        if (!Particle.connected()) {
            Log.error("Could not connect!");
            return;
        }
  }
    
}
```

### References


---

### Completeness
